scripts/eurostat/health_determinants/Alcohol_Consumption/process_test.py

```diff
@@ -0,0 +1,101 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Script to automate the testing for USA Population by Race preprocess script.
+"""
+
+import os
+from os import path
+import unittest
+from process import EuroStatAlcoholConsumption
+# module_dir_ is the path to where this test is running from.
+module_dir_ = os.path.dirname(__file__)
+test_data_folder = os.path.join(module_dir_, "test_data")
+op_data_folder = os.path.join(module_dir_, "test_output_data")
+input_path = os.path.dirname(os.path.abspath(__file__)) + os.sep + "input_data"
+
+
+class TestPreprocess(unittest.TestCase):
+    """
+    TestPreprocess is inherting unittest class
+    properties which further requried for unit testing
+    """
+    cleaned_csv_file_path = os.path.join(op_data_folder, "data.csv")
+    mcf_file_path = os.path.join(op_data_folder, "test_census.mcf")
+    tmcf_file_path = os.path.join(op_data_folder, "test_census.tmcf")
+
+    ip_data = [
+        os.path.dirname(os.path.abspath(__file__)) + os.sep +
+        'test_data/hlth_ehis_al1e.tsv',
+        os.path.dirname(os.path.abspath(__file__)) + os.sep +
+        'test_data/hlth_ehis_de6.tsv'
+    ]
+    base = EuroStatAlcoholConsumption(ip_data, cleaned_csv_file_path,
+                                    mcf_file_path, tmcf_file_path)
+    base.process()
+
+    def test_mcf_tmcf_files(self):
+        """
+        This method is required to test between output generated
+        preprocess script and excepted output files like MCF File
+        """
+        expected_mcf_file_path = os.path.join(
+            test_data_folder,
+            "expected_eurostat_population_alcoholconsumption.mcf")
+
+        expected_tmcf_file_path = os.path.join(
+            test_data_folder,
+            "expected_eurostat_population_alcoholconsumption.tmcf")
+
+        with open(expected_mcf_file_path,
+                  encoding="UTF-8") as expected_mcf_file:
+            expected_mcf_data = expected_mcf_file.read()
+
+        with open(expected_tmcf_file_path,
+                  encoding="UTF-8") as expected_tmcf_file:
+            expected_tmcf_data = expected_tmcf_file.read()
+
+        with open(self.mcf_file_path, encoding="UTF-8") as mcf_file:
+            mcf_data = mcf_file.read()
+
+        with open(self.tmcf_file_path, encoding="UTF-8") as tmcf_file:
+            tmcf_data = tmcf_file.read()
+        if path.exists(self.mcf_file_path):
+            os.remove(self.mcf_file_path)
+        if path.exists(self.tmcf_file_path):
+            os.remove(self.tmcf_file_path)
```

Comment:
Please use tempfile module to avoid os operations for temporary files / folder creations.
Refer: https://github.com/datacommonsorg/data/search?q=tempfile